bft_types/src/lib.rs

```diff
@@ -220,23 +225,28 @@ impl BfProgram {
     ///# }
     /// ```
     pub fn analyse_program(&mut self) -> Result<(), String> {
+        // Ensure we don't double the jump map if analyse gets called twice.
+        // Might be better as a flag in the struct?
```

Comment:
Make this non-public and call it from your initialisation methods and this wouldn't happen.